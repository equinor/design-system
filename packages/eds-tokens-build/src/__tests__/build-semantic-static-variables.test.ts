import { describe, it, expect, beforeEach, afterEach } from 'vitest'
import { readFileSync, existsSync } from 'fs'
import path from 'path'
import {
  setupTestWorkspace,
  cleanupTestWorkspace,
  runScripts,
} from './test-utils'

describe('Semantic Static Variables Build', () => {
  let testWorkspace: string

  beforeEach(() => {
    testWorkspace = setupTestWorkspace()
  })

  afterEach(() => {
    cleanupTestWorkspace(testWorkspace)
  })

  it('should build CSS static variables from semantic tokens', () => {
    // Run full pipeline up to static variables
    runScripts(
      [
        'generate-color-scheme-tokens.js',
        'generate-concept-tokens.js',
        'generate-semantic-tokens.js',
        'generate-dynamic-appearance-tokens.js',
        'build-color-scheme-variables.js',
        'build-semantic-static-variables.js',
      ],
      testWorkspace,
    )

    // Verify static CSS was generated
    const staticCSS = path.join(
      testWorkspace,
      'build/css/color/static/variables.css',
    )
    expect(existsSync(staticCSS)).toBe(true)

    // Snapshot test the generated CSS
    const actualStatic = readFileSync(staticCSS, 'utf-8')
    expect(actualStatic).toMatchSnapshot('static-variables.css')
  })
})
