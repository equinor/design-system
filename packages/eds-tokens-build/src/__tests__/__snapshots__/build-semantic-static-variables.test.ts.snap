// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Semantic Static Variables Build > should build CSS static variables from semantic tokens > static-variables.css 1`] = `
"/**
 * Do not edit directly, this file was auto-generated by scripts/build-static-variables.ts.
 * Note: color scheme variables have moved to color-scheme.css. Import both files.
 */

:root {
  --x-color-bg-accent-canvas: var(--x-color-orange-1);
  --x-color-bg-accent-surface: var(--x-color-orange-2);
  --x-color-bg-accent-fill-muted-default: var(--x-color-orange-3);
  --x-color-bg-accent-fill-muted-hover: var(--x-color-orange-4);
  --x-color-bg-accent-fill-muted-active: var(--x-color-orange-5);
  --x-color-bg-accent-fill-emphasis-default: var(--x-color-orange-9);
  --x-color-bg-accent-fill-emphasis-hover: var(--x-color-orange-10);
  --x-color-bg-accent-fill-emphasis-active: var(--x-color-orange-11);
  --x-color-bg-neutral-canvas: var(--x-color-gray-1);
  --x-color-bg-neutral-surface: var(--x-color-gray-2);
  --x-color-bg-neutral-fill-muted-default: var(--x-color-gray-3);
  --x-color-bg-neutral-fill-muted-hover: var(--x-color-gray-4);
  --x-color-bg-neutral-fill-muted-active: var(--x-color-gray-5);
  --x-color-bg-neutral-fill-emphasis-default: var(--x-color-gray-9);
  --x-color-bg-neutral-fill-emphasis-hover: var(--x-color-gray-10);
  --x-color-bg-neutral-fill-emphasis-active: var(--x-color-gray-11);
  --x-color-bg-info-canvas: var(--x-color-blue-1);
  --x-color-bg-info-surface: var(--x-color-blue-2);
  --x-color-bg-info-fill-muted-default: var(--x-color-blue-3);
  --x-color-bg-info-fill-muted-hover: var(--x-color-blue-4);
  --x-color-bg-info-fill-muted-active: var(--x-color-blue-5);
  --x-color-bg-info-fill-emphasis-default: var(--x-color-blue-9);
  --x-color-bg-info-fill-emphasis-hover: var(--x-color-blue-10);
  --x-color-bg-info-fill-emphasis-active: var(--x-color-blue-11);
  --x-color-bg-success-canvas: var(--x-color-green-1);
  --x-color-bg-success-surface: var(--x-color-green-2);
  --x-color-bg-success-fill-muted-default: var(--x-color-green-3);
  --x-color-bg-success-fill-muted-hover: var(--x-color-green-4);
  --x-color-bg-success-fill-muted-active: var(--x-color-green-5);
  --x-color-bg-success-fill-emphasis-default: var(--x-color-green-9);
  --x-color-bg-success-fill-emphasis-hover: var(--x-color-green-10);
  --x-color-bg-success-fill-emphasis-active: var(--x-color-green-11);
  --x-color-bg-warning-canvas: var(--x-color-orange-1);
  --x-color-bg-warning-surface: var(--x-color-orange-2);
  --x-color-bg-warning-fill-muted-default: var(--x-color-orange-3);
  --x-color-bg-warning-fill-muted-hover: var(--x-color-orange-4);
  --x-color-bg-warning-fill-muted-active: var(--x-color-orange-5);
  --x-color-bg-warning-fill-emphasis-default: var(--x-color-orange-9);
  --x-color-bg-warning-fill-emphasis-hover: var(--x-color-orange-10);
  --x-color-bg-warning-fill-emphasis-active: var(--x-color-orange-11);
  --x-color-bg-danger-canvas: var(--x-color-red-1);
  --x-color-bg-danger-surface: var(--x-color-red-2);
  --x-color-bg-danger-fill-muted-default: var(--x-color-red-3);
  --x-color-bg-danger-fill-muted-hover: var(--x-color-red-4);
  --x-color-bg-danger-fill-muted-active: var(--x-color-red-5);
  --x-color-bg-danger-fill-emphasis-default: var(--x-color-red-9);
  --x-color-bg-danger-fill-emphasis-hover: var(--x-color-red-10);
  --x-color-bg-danger-fill-emphasis-active: var(--x-color-red-11);
  --x-color-border-accent-subtle: var(--x-color-orange-6);
  --x-color-border-accent-medium: var(--x-color-orange-7);
  --x-color-border-accent-strong: var(--x-color-orange-8);
  --x-color-border-neutral-subtle: var(--x-color-gray-6);
  --x-color-border-neutral-medium: var(--x-color-gray-7);
  --x-color-border-neutral-strong: var(--x-color-gray-8);
  --x-color-border-info-subtle: var(--x-color-blue-6);
  --x-color-border-info-medium: var(--x-color-blue-7);
  --x-color-border-info-strong: var(--x-color-blue-8);
  --x-color-border-success-subtle: var(--x-color-green-6);
  --x-color-border-success-medium: var(--x-color-green-7);
  --x-color-border-success-strong: var(--x-color-green-8);
  --x-color-border-warning-subtle: var(--x-color-orange-6);
  --x-color-border-warning-medium: var(--x-color-orange-7);
  --x-color-border-warning-strong: var(--x-color-orange-8);
  --x-color-border-danger-subtle: var(--x-color-red-6);
  --x-color-border-danger-medium: var(--x-color-red-7);
  --x-color-border-danger-strong: var(--x-color-red-8);
  --x-color-text-accent-subtle: var(--x-color-orange-12); /** Used for text and icons */
  --x-color-text-accent-strong: var(--x-color-orange-13); /** Used for text and icons */
  --x-color-text-accent-subtle-on-emphasis: var(--x-color-orange-14); /** Text or icons against colored backgrounds */
  --x-color-text-accent-strong-on-emphasis: var(--x-color-orange-15); /** Text or icons against colored backgrounds */
  --x-color-text-neutral-subtle: var(--x-color-gray-12); /** Used for text and icons */
  --x-color-text-neutral-strong: var(--x-color-gray-13); /** Used for text and icons */
  --x-color-text-neutral-subtle-on-emphasis: var(--x-color-gray-14); /** Text or icons against colored backgrounds */
  --x-color-text-neutral-strong-on-emphasis: var(--x-color-gray-15); /** Text or icons against colored backgrounds */
  --x-color-text-info-subtle: var(--x-color-blue-12); /** Used for text and icons */
  --x-color-text-info-strong: var(--x-color-blue-13); /** Used for text and icons */
  --x-color-text-info-subtle-on-emphasis: var(--x-color-blue-14); /** Text or icons against colored backgrounds */
  --x-color-text-info-strong-on-emphasis: var(--x-color-blue-15); /** Text or icons against colored backgrounds */
  --x-color-text-success-subtle: var(--x-color-green-12); /** Used for text and icons */
  --x-color-text-success-strong: var(--x-color-green-13); /** Used for text and icons */
  --x-color-text-success-subtle-on-emphasis: var(--x-color-green-14); /** Text or icons against colored backgrounds */
  --x-color-text-success-strong-on-emphasis: var(--x-color-green-15); /** Text or icons against colored backgrounds */
  --x-color-text-warning-subtle: var(--x-color-orange-12); /** Used for text and icons */
  --x-color-text-warning-strong: var(--x-color-orange-13); /** Used for text and icons */
  --x-color-text-warning-subtle-on-emphasis: var(--x-color-orange-14); /** Text or icons against colored backgrounds */
  --x-color-text-warning-strong-on-emphasis: var(--x-color-orange-15); /** Text or icons against colored backgrounds */
  --x-color-text-danger-subtle: var(--x-color-red-12); /** Used for text and icons */
  --x-color-text-danger-strong: var(--x-color-red-13); /** Used for text and icons */
  --x-color-text-danger-subtle-on-emphasis: var(--x-color-red-14); /** Text or icons against colored backgrounds */
  --x-color-text-danger-strong-on-emphasis: var(--x-color-red-15); /** Text or icons against colored backgrounds */
}
"
`;
